src/cditools/eiger_async.py

```diff
@@ -0,0 +1,685 @@
+"""
+Ophyd Async implementation for Eiger detector.
+"""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import AsyncGenerator, AsyncIterator, Sequence
+from logging import getLogger
+from pathlib import Path
+from typing import Annotated as A
+from typing import Any, Literal
+from urllib.parse import urlunparse
+
+import numpy as np  # type: ignore[import-not-found]
+from bluesky.protocols import StreamAsset
+from event_model import (  # type: ignore[import-untyped]
+    ComposeStreamResource,
+    ComposeStreamResourceBundle,
+    DataKey,  # type: ignore[import-untyped]
+)
+from ophyd_async.core import (
+    DetectorTrigger,
+    HDFDatasetDescription,
+    HDFDocumentComposer,
+    PathInfo,
+    PathProvider,
+    SignalDatatypeT,
+    SignalR,
+    SignalRW,
+    StrictEnum,
+    TriggerInfo,
+    observe_value,
+)
+from ophyd_async.epics.adcore import (
+    ADBaseController,
+    ADBaseDatasetDescriber,
+    ADBaseIO,
+    ADFileWriteMode,
+    ADImageMode,
+    ADWriter,
+    AreaDetector,
+    NDArrayBaseIO,
+    NDPluginBaseIO,
+)
+from ophyd_async.epics.signal import PvSuffix
+
+logger = getLogger(__name__)
+
+
+# TODO: Port to ophyd-async
+# ==============================================================================
+class HDFDatasetDescription2(HDFDatasetDescription):
+    join_method: Literal["stack", "concat"] = "concat"
+
+
+class HDFDocumentComposer2(HDFDocumentComposer):
```

Comment:
Are these `2` names for these classes placeholders?

Reply:
Actually these were temporary workarounds for testing the "join_method" parameter. I can remove them.

Reply:
Ah the new ophyd-async version breaks this for me :cry: 

Looks like I will have to make my own composer